src/_P001_Switch.ino

```diff
@@ -655,14 +740,14 @@ boolean Plugin_001(byte function, struct EventStruct *event, String& string)
                 log += CONFIG_PIN1;
                 log += F(" State=");
                 log += state ? '1' : '0';
-                log += F(" Output value=");
+              	log += F(" Output=");
                 log += output_value;
                 addLog(LOG_LEVEL_INFO, log);
               }
               sendData(event);
 
               // reset Userdata so it displays the correct state value in the web page
-              UserVar[event->BaseVarIndex] = sendState ? 1 : 0;
+//              UserVar[event->BaseVarIndex] = sendState ? 1 : 0;
```

Comment:
and here?

Reply:
With the flashing detection the state of the UserVar[event->BaseVarIndex] can be
0, 1 or 2 (flashing)!
A few lines above the commented line you can find the code
UserVar[event->BaseVarIndex] = output_value;

The commented line can be deleted, it was just a reminder for me.

Reply:
You can't remove that line.
Its meant for displaying the state in the pinstate page.

Reply:
And in the device page

Reply:
The state is saved a few lines above, see attached picture
![Highlighted code](https://user-images.githubusercontent.com/55511593/67765790-3af3e900-fa4d-11e9-8b8a-253c60f152c7.JPG)

Reply:
I developed the logic one year ago, so I had to review it.

Let me explain the logic:
for example, the event=3 means double-click, but it's not a valid pin state.
Example: the device generates event=3 (double-click) and send the event to the Rules with the command "sendData(event)".
The UserVar is used to pass the event number to the Rules, but it's not necessarily a valid pinstate value (0 or 1 ONLY).
Therefore at the end of the procedure you should set the UserVar[event->BaseVarIndex] = the state of the pin (either 0 or 1).
It cannot be 3, otherwise you will see "3" in the pinstate page.

Is it clear?